src/routes/messages/entities/message-item.entity.ts

```diff
@@ -0,0 +1,12 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { Message } from './message.entity';
+
+export class MessageItem extends Message {
+  @ApiProperty()
+  type: string;
+
+  constructor(...args: ConstructorParameters<typeof Message>) {
+    super(...args);
+    this.type = 'MESSAGE';
+  }
```

Comment:
I would leave the type when we have a collection of different types (eg.: collection of dates and messages).

Adding a `type` in other message related endpoints is not an API breaking change but semantically it might not make sense. If you hit a resource where you are expected to only get messages why is a `MESSAGE` type returned?

This should be dealt within the service in my opinion. We should be able to reuse entities and show/hide properties when required.

Reply:
Yes, I agree, having the `type: MESSAGE` property in other related endpoint responses it's maybe doubtful. So I think we could keep this as it is and use this `MessageItem` class to map the expected format. What do you think?

Another option would be adding the `type: MESSAGE` property in the service and using `Required<Message>` (https://www.typescriptlang.org/docs/handbook/utility-types.html#requiredtype) but it's maybe a bit overkill for this use case.